cmd/oras/root/attach.go

```diff
@@ -180,9 +185,12 @@ func runAttach(cmd *cobra.Command, opts *attachOptions) error {
 	if !strings.HasSuffix(opts.RawReference, digest) {
 		opts.RawReference = fmt.Sprintf("%s@%s", opts.Path, subject.Digest)
 	}
-	fmt.Println("Attached to", opts.AnnotatedReference())
-	fmt.Println("Digest:", root.Digest)
+	display.Print("Attached to", opts.AnnotatedReference())
+	display.Print("Digest:", root.Digest)
 
 	// Export manifest
-	return opts.ExportManifest(ctx, store, root)
+	if err = opts.ExportManifest(ctx, store, root); err != nil {
+		return err
+	}
+	return opts.WriteTo(os.Stdout, meta.NewPush(root, opts.Path))
```

Comment:
The high level design should be something like
```mermaid
flowchart LR
    core["Core Process"] -- streaming --> object["Structured\nOutput\nObjects"]
    object -- streaming --> display["Display\nEvent\nHandler"]
```
Turning the above design into code. It will be
```go
// create an event handler for `oras attach`
handler := display.NewAttachHandler("text") // can also be json or others

doAttachCore()

// Publish display event
handler.OnAttached(root, opts) // event args can be typed
```
A handler can be implemented as
```go
type textAttachHandler struct{}

func (textAttachHandler) OnAttached(root ocispec.Descriptor, opts *option.Target) error {
	fmt.Println("Attached to", opts.AnnotatedReference())
	fmt.Println("Digest:", root.Digest)
	return nil
}
```

Reply:
You may also consider the event pattern:
- https://learn.microsoft.com/dotnet/standard/events/
- https://learn.microsoft.com/dotnet/csharp/modern-events
- https://learn.microsoft.com/dotnet/standard/design-guidelines/event

Reply:
Since we are coding in golang, you may consider use `chan` to simulate the `event` but I think it perplexes stuffs.

Reply:
I can see two suggested changes here
1) push metadata should only be assemble when `--format` flag is not nil
2) formalize text status output and metadata output to an event-based handler

The second change is hard since text output is high-coupled with initializing copy option struct.